tests/test_paperqa.py

```diff
@@ -1239,6 +1240,70 @@ def test_dois_resolve_to_correct_journals(doi_journals):
     assert details.journal == doi_journals["journal"]
 
 
+def test_docdetails_merge_with_non_list_fields():
+    # Test merging a DocDetail that was republished
+    initial_date = datetime(2023, 1, 1)
+    doc1 = DocDetails(
+        citation="Citation 1",
+        publication_date=initial_date,
+        docname="Document 1",
+        dockey="key1",
+        other={"bibtex_source": "source1", "client_source": ["client1"]},
+    )
+
+    later_publication_date = initial_date + timedelta(weeks=13)
+    doc2 = DocDetails(
+        citation=doc1.citation,
+        publication_date=later_publication_date,
+        docname=doc1.docname,
+        dockey=doc1.dockey,
+        other={"bibtex_source": ["source2"], "client_source": "client2"},
```

Comment:
```suggestion
        # NOTE: doc1 has non-list bibtex_source and list client_source
        other={"bibtex_source": ["source2"], "client_source": "client2"},
```

Let's document the test